watcher/watcher.go

```diff
@@ -0,0 +1,156 @@
+package watcher
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/Tonkpils/snag/exchange"
+	fsn "gopkg.in/fsnotify.v1"
+)
+
+const rebuild = "rebuild"
+
+type Watcher interface {
+	Watch(string) error
+}
+
+type FSWatcher struct {
+	ex           exchange.SendListener
+	fsn          *fsn.Watcher
+	done         chan struct{}
+	mtimes       map[string]time.Time
+	watching     map[string]struct{}
+	watchDir     string
+	ignoredItems []string
+}
+
+func New(ex exchange.SendListener, ignoredItems []string) (Watcher, error) {
```

Comment:
My only concern again is each component knowing what events they will be triggering. Again thinking of separating these things into packages why should the watcher know that there's an event called `rebuild` being listened to on the builder. In any case, for the sake of keeping PRs short and sweet, I think this is a good enough restructure and we'll iterate over it. I'll code up what I mean on my arguments and if we like it we can always merge it or close it if not.